src/features/Layout/components/parts/SearchInputComponent.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useState } from 'react';
+import { debounce } from "lodash";
+import { useAppSelector, useAppDispatch } from '../../../../app/hooks';
+import { getCharacterFilteredList, setLastUrlQuery, setSearch, setSearched } from '../../../redux-slices/characterSlice';
+import {
+    generateUrl
+} from '../../../utils';
+
+interface Props{
+    className: string;
+    placeholder: string;
+}
+
+const SearchInputComponent = ({className, placeholder}: Props) =>
+{
+    const dispatch = useAppDispatch();
+    const optionList = useAppSelector(state => state.dropdown.optionList);
+    const lastQuery = useAppSelector(state => state.character.search);
+    const searched = useAppSelector(state => state.character.searched)
+    const [query, setQuery] = useState('');
+
+    const getData = async (query: string) =>
+    {   
+        
+    }
+
+    const debouncedSearch = debounce(async (query) => {
```

Comment:
A custom 10LOC implementation for debounce could help you remove lodash lib altogether